src/main/java/org/folio/edge/oaipmh/Verb.java

```diff
@@ -0,0 +1,152 @@
+package org.folio.edge.oaipmh;
+
+import io.vertx.ext.web.RoutingContext;
+import org.openarchives.oai._2.OAIPMHerrorType;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableSet.of;
+import static java.util.Collections.EMPTY_SET;
+import static java.util.stream.Collectors.toSet;
+import static java.util.stream.Stream.concat;
+import static org.folio.edge.core.Constants.PARAM_API_KEY;
+import static org.folio.edge.oaipmh.Constants.FROM;
+import static org.folio.edge.oaipmh.Constants.IDENTIFIER;
+import static org.folio.edge.oaipmh.Constants.METADATA_PREFIX;
+import static org.folio.edge.oaipmh.Constants.RESUMPTION_TOKEN;
+import static org.folio.edge.oaipmh.Constants.SET;
+import static org.folio.edge.oaipmh.Constants.UNTIL;
+import static org.folio.edge.oaipmh.Constants.VERB;
+import static org.openarchives.oai._2.OAIPMHerrorcodeType.BAD_ARGUMENT;
+
+/**
+ * Enum that represents OAI-PMH verbs with associated http parameters and validation logic.
+ */
+public enum Verb {
+  GET_RECORD("GetRecord", of(IDENTIFIER, METADATA_PREFIX), EMPTY_SET, null),
```

Comment:
I'm not familiar with Guava, so this is new to me... pretty convenient!

Reply:
Yeah, Java 8 still missing pretty known and convenient features. Java 9 though introduced these convenience factory methods for collections.